### PR TITLE
#3023 Add Counts to each tabs in CodeTrek 

### PR DESCRIPTION
**Pull request by jaskaransingh304**
Targets #3023 

### Description

Several changes were implemented in Codetrek to enhance the user experience. Firstly, the service page of Codetrek underwent modifications to display the total counts alongside the tabs. This improvement allows users to quickly ascertain the number of items associated with each tab.

In addition, adjustments were made to the controller to pass the necessary requests to the view page. This ensures that the updated functionality is properly integrated and accessible to users.

Moreover, variable values were assigned adjacent to the status names in the view page. As a result, these values are displayed beside each tab, providing users with immediate information about the associated status.

Overall, these changes in Codetrek's service page, controller, and view page contribute to an improved user interface and facilitate easier navigation and understanding of the data.

### Checklist:

- [x] I have performed a self-review of my own code.

### Approach -
Checked the view page and tried to figure , how data can be passed in it.
Checked the view page of projects , and figured how function was created in service page and used in menu_header. Took its reference. 

### Code review and documentation: 1 hour.
then ,
### Changes required in code -
As there was no function to count status, so added a function in Codetrek Service page.
Return request was made in controller as what it will return to view page, then added the counts next to the status name.

### CODE REVIEW

File: Modules/CodeTrek/Http/Controllers/CodeTrekController.php [hunk 1]
```diff
@@ -27,9 +27,21 @@ public function __construct(CodeTrekService $service)
     public function index(Request $request)
     {
         $centres = OfficeLocation::all();
+        $applicantData = $this->service->getCodeTrekApplicants($request->all());
+        $applicants = $applicantData['applicants'];
+        $activeCount = $applicantData['activeCount'];
+        $inactiveCount = $applicantData['inactiveCount'];
+        $completedCount = $applicantData['completedCount'];
```
Comment (GauravGusain98) [inline Modules/CodeTrek/Http/Controllers/CodeTrekController.php:34]: These are unwanted variables. We can simply return the $applicantsData variable to view

File: Modules/CodeTrek/Http/Controllers/CodeTrekController.php [hunk 2]
```diff
@@ -27,9 +27,21 @@ public function __construct(CodeTrekService $service)
     public function index(Request $request)
     {
         $centres = OfficeLocation::all();
+        $applicantData = $this->service->getCodeTrekApplicants($request->all());
+        $applicants = $applicantData['applicants'];
```
Comment (GauravGusain98) [inline Modules/CodeTrek/Http/Controllers/CodeTrekController.php:31]: I think we are getting applicants data and not only applicants. So lets change the variable name to $applicantsData

File: Modules/CodeTrek/Services/CodeTrekService.php [hunk 3]
```diff
@@ -23,7 +23,16 @@ public function getCodeTrekApplicants($data = [])
         ->paginate(config('constants.pagination_size'))
         ->appends(request()->except('page'));
 
-        return ['applicants' => $applicants];
+        $activeCount = CodeTrekApplicant::where('status', 'active')->count();
+        $inactiveCount = CodeTrekApplicant::where('status', 'inactive')->count();
+        $completedCount = CodeTrekApplicant::where('status', 'completed')->count();
```
Comment (GauravGusain98) [inline Modules/CodeTrek/Services/CodeTrekService.php:28]: Can you re write this using group function such that you will get 2 columns - status_name and count of applicants in that status. 

You can read about group clause in my SQL and see if laravel provides any function for grouping.

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]

File: Modules/CodeTrek/Services/CodeTrekService.php (hunk 3 above)
Comment (jaskaransingh304) [reply Modules/CodeTrek/Services/CodeTrekService.php:28]: I will do that

File: Modules/CodeTrek/Http/Controllers/CodeTrekController.php (hunk 1 above)
Comment (jaskaransingh304) [reply Modules/CodeTrek/Http/Controllers/CodeTrekController.php:34]: Sure

**Comment by jaskaransingh304** [comment]
image
image
     
Attaching screenshot for reference

File: Modules/CodeTrek/Services/CodeTrekService.php [hunk 4]
```diff
@@ -23,7 +23,30 @@ public function getCodeTrekApplicants($data = [])
...
+        $statusCounts = [];
+
+        $statusValues = ['active', 'inactive', 'completed'];
+        foreach ($statusValues as $status) {
+            $statusCounts[$status] = 0;
+        }
+
+        foreach ($applicantsData as $data) {
+            $statusCounts[$data->status] = $data->total;
+        }
```
Comment (GauravGusain98) [inline Modules/CodeTrek/Services/CodeTrekService.php:43]: The the slug valuesof status from config.php which is inside `Modules/CodeTrek/config`
image

Comment (GauravGusain98) [reply Modules/CodeTrek/Services/CodeTrekService.php:43]: Here take the key names from config
```suggestion
        $statusCounts = [
            'active' => 0,
            'inactive' => 0,
            'completed' => 0
        ];
        foreach ($applicantsData as $data) {
            $statusCounts[$data->status] = $data->total;
        }
```

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]

**Review by GauravGusain98 — APPROVED** [review]

**Pull request merged**